allennlp/run.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Dict, Union
```

Comment:
In my opinion releasing a pip package should be our primary goal for distributing allennlp (and how we manage supported releases), with exact reproducibility with Docker an important second. Most of the stuff for making a pip package is already there in `setup.py`, we just need to update some dependencies and emails and stuff.